### PR TITLE
Fix #175

### LINKED ISSUE #175: Inserts with timestamp parameters raise error
https://github.com/uber/vertica-python/blob/37e6b9cd33a2eec905f2b4b1f00e4cc004aff383/vertica_python/vertica/cursor.py#L348

Line 348 is adding timestamp bind parameters to my INSERT statement's VALUES clause without surrounding quotes.

e.g. ```INSERT INTO some_table (some_timestamp) VALUES (2016-05-27 16:52:55.658182)```

should become

```INSERT INTO some_table (some_timestamp) VALUES ('2016-05-27 16:52:55.658182')```

### PR DESCRIPTION
**Pull request by darwinyip**
Fixes #175 about timestamps not being surrounded by quotes when inserting.

### CODE REVIEW

File: vertica_python/vertica/cursor.py [hunk 1]
```diff
@@ -377,6 +378,8 @@ def format_operation_with_parameters(self, operation, parameters, is_csv=False):
 
                 if isinstance(param, string_types):
                     param = self.format_quote(as_text(param), is_csv)
+                elif isinstance(param, datetime.datetime):
```
Comment (sitingren) [inline vertica_python/vertica/cursor.py]: Same logic should be added under line 398.

Comment (darwinyip) [reply vertica_python/vertica/cursor.py]: Done

Comment (sitingren) [reply vertica_python/vertica/cursor.py]: Thanks. There are three datatypes that need to be quoted: DATE(datetime.date), DATETIME(datetime.datetime), and TIME(datetime.time).

**Comment by sitingren** [comment]
Thank you for your contribution, please also add tests before we can merge.

**Comment by darwinyip** [comment]
@sitingren I want to add a test to `test_cursor.py` but need to create a new table or add a new timestamp field to `cursor_test` table. What is the recommended approach?

File: vertica_python/tests/integration_tests/test_cursor.py [hunk 2]
```diff
@@ -401,6 +402,30 @@ def test_format_quote_unicode(self):
...
+            cur.execute("""CREATE TABLE {0} (
+                                a INT,
+                                b VARCHAR(32),
+                                c TIMESTAMP,
+                                d DATE
+                           )
+                        """.format(self._table))
+
+            cur.execute("INSERT INTO {0} (a, b, c, d) VALUES (:n, :s, :t, :d)".format(self._table),
+                            {'n': 10, 's': 'aa', 't': datetime.datetime(2018, 9, 7, 15, 38, 19, 769000), 'd': datetime.date(2018, 9, 7)})
```
Comment (sitingren) [inline vertica_python/tests/integration_tests/test_cursor.py]: The test looks good. Feel free to reuse `self._table` or create your new table, as long as you clean up tables at the end.

Last thing, don't forget to add a `TIME` column. Then you are good to go :)

**Comment by darwinyip** [comment]
Any reason why keep time as a string?

**Comment by sitingren** [comment]
I guess contributors haven't implemented the TIME parser yet and just leave TIME data as string.

Thank you for your contribution again!

**Comment by darwinyip** [comment]
Thanks @sitingren . By the way, when can we expect the next release? And will it simultaneously get pushed to `pip`?

**Comment by sitingren** [comment]
@darwinyip We would like to include #216 Travis CI setup to the next release, this is an important work we think we must have. We may also fix some bugs (e.g. #212) if we have time.

And yes, the release will be published to both [the GitHub page](https://github.com/vertica/vertica-python/releases) and PyPI (so you can run `pip install vertica-python`).

**Pull request merged**